src/recommendationservice/recommendation_server.py

```diff
@@ -127,30 +145,38 @@ def check_feature_flag(flag_name: str):
 
 
 if __name__ == "__main__":
-    service_name = must_map_env('OTEL_SERVICE_NAME')
+
+    # service_name = must_map_env('OTEL_SERVICE_NAME')
     api.set_provider(FlagdProvider(host=os.environ.get('FLAGD_HOST', 'flagd'), port=os.environ.get('FLAGD_PORT', 8013)))
-    api.add_hooks([TracingHook()])
+    # api.add_hooks([TracingHook()])
 
     # Initialize Traces and Metrics
-    tracer = trace.get_tracer_provider().get_tracer(service_name)
-    meter = metrics.get_meter_provider().get_meter(service_name)
-    rec_svc_metrics = init_metrics(meter)
+    # tracer = trace.get_tracer_provider().get_tracer(service_name)
+    # meter = metrics.get_meter_provider().get_meter(service_name)
+    # rec_svc_metrics = init_metrics(meter)
 
     # Initialize Logs
-    logger_provider = LoggerProvider(
-        resource=Resource.create(
-            {
-                'service.name': service_name,
-            }
-        ),
-    )
-    set_logger_provider(logger_provider)
-    log_exporter = OTLPLogExporter(insecure=True)
-    logger_provider.add_log_record_processor(BatchLogRecordProcessor(log_exporter))
-    handler = LoggingHandler(level=logging.NOTSET, logger_provider=logger_provider)
-
-    # Attach OTLP handler to logger
+    # logger_provider = LoggerProvider(
+    #     resource=Resource.create(
+    #         {
+    #             'service.name': service_name,
+    #         }
+    #     ),
+    # )
+    # set_logger_provider(logger_provider)
+    # log_exporter = OTLPLogExporter(insecure=True)
+    # logger_provider.add_log_record_processor(BatchLogRecordProcessor(log_exporter))
+    # handler = LoggingHandler(level=logging.NOTSET, logger_provider=logger_provider)
+
+    # Logging
     logger = logging.getLogger('main')
+    logger.setLevel(logging.INFO)
+
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setLevel(logging.INFO)
+
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
```

Comment:
With the otel instrumentation logs were decorated with trace context. I'm pretty certain how you have things set up with the NR python agent this won't happen. Follow the NR docs https://docs.newrelic.com/docs/logs/logs-context/configure-logs-context-python/